homeassistant/components/rako/strings.json

```diff
@@ -0,0 +1,19 @@
+{
+    "title": "Rako",
+    "config": {
+        "error": {
+            "no_devices_found": "No Bridge found on the network",
+            "cannot_connect": "Cannot connect to the Bridge"
+        },
+        "step": {
+            "user": {
+                "title": "Rako Bridge Configuration",
+                "description": "If pre-populated, Bridge was auto-discovered!",
+                "data": {
+                    "host": "Host IP",
+                    "port": "Port"
```

Comment:
Use key references where possible.

Reply:
hey @milanmeu - so i see people [using keys like this](https://github.com/home-assistant/core/blob/f516650ab7c1e900cd437b845a2fc04b09d6e217/homeassistant/components/hue/strings.json#L13), but its black magic to me. could you point me to a list of available keys / documentation about how it works?

Reply:
It is recommended to use [key references](https://docs.lokalise.com/en/articles/1400528-key-referencing) so that we only have to translate common words once in [Lokalise](https://lokalise.com/signup/130246255a974bd3b5e8a1.51616605/all/).
Developers docs: https://developers.home-assistant.io/docs/translations#translation-placeholders

### Common keys
The `common` keys list: [homeassistant/strings.json](https://github.com/home-assistant/core/blob/dev/homeassistant/strings.json)
These have the following format: `[%key:common::JSON-PATH::JSON-PATH::THE-KEY%]`
Example:
`[%key:common::config_flow::data::port%]` will be converted to the value from [homeassistant/strings.json](https://github.com/home-assistant/core/blob/dev/homeassistant/strings.json): `Port`.


### Component keys
It is also possible to make a reference to a key in [homeassistant/components/rako/strings.json](https://github.com/home-assistant/core/pull/45915/files/f516650ab7c1e900cd437b845a2fc04b09d6e217#diff-12b41571ac481723939fddc1f22894804eab5959adcaa7c4f56c5c2432fc8099).
These have the following format: `[%key:component::rako::JSON-PATH::JSON-PATH::key%]`
Example: 
`[%key:component::rako::config::step::user::description%]` will be converted to `If pre-populated, Bridge was auto-discovered!`

### More information
It is possible to combine references with strings. Example:
`[%key:common::config_flow::data::name%] (Optional)` becomes `Name (Optional)`

It is possible to combine references with references. Example:
`[%key:common::config_flow::data::host%]:[%key:common::config_flow::data::port%]` becomes `Host:Port` 

It is not allowed to make a reference to a reference. (https://github.com/home-assistant/core/pull/46999)
It is not allowed to make a component reference to another integration.

### Suggestion

- `Port` -> `[%key:common::config_flow::data::port%]` (Port)

Choose what fits best or don't use a reference if none fits:
- `Host IP` -> `[%key:common::config_flow::data::host%]` (Host)
- `Host IP` -> `[%key:common::config_flow::data::ip%]` (IP Address)
- `Host IP` -> `[%key:common::config_flow::data::host%] or [%key:common::config_flow::data::ip%]` (Host or IP Address)

This changes "bridge" to "device". If bridge fits better then no reference can be used.
- `No bridge found on the network` -> `[%key:common::config_flow::abort::no_devices_found%]` (No device found on the network)

Choose what fits best or don't use a reference if none fits:
- `Cannot connect to the Bridge` -> `[%key:common::config_flow::error::cannot_connect%]` (Failed to connect)
- `Cannot connect to the Bridge` -> `[%key:common::config_flow::error::cannot_connect%]` (Failed to connect to the Bridge)

Reply:
thanks, thats really helpful. i totally missed that whole part of the dev docs!